packages/shared/lib/core/wallet/utils/outputs/preprocessGroupedOutputs.ts

```diff
@@ -46,3 +48,23 @@ function isTransactionIncoming(outputs: OutputData[], accountAddress: string): b
     const address = getRecipientAddressFromOutput(nonRemainderOutputs[0].output)
     return address === accountAddress
 }
+
+function splitOutputs(outputDatas: OutputData[]): IOutput[][] {
+    const outputs = outputDatas.map((outputData) => ({
+        outputId: outputData.outputId,
+        output: outputData.output,
+    }))
+
+    const aliasOutput = outputs.find((output) => output.output.type === OUTPUT_TYPE_ALIAS)
+    const containsNewAliasOutput = (aliasOutput?.output as IAliasOutput)?.stateIndex === 0
+
+    if (containsNewAliasOutput) {
+        const aliasOutputIndex = outputs.findIndex((output) => output.output.type === OUTPUT_TYPE_ALIAS)
```

Comment:
this is only needed if we have tx which contain the foundry AND alias. Not sure if i should let this in